ReactNativeClient/lib/services/filterParser.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-cond-assign */
+
+// match[1] AND/OR tagName:tagValue
+// match[2] AND/OR
+// match[3] -
+// match[4] tagName
+// match[6] tagValue
+// const filterRegex = /(\b(AND|OR)\s+)?(-?)([\w]+|"[^"]+")\s*(:\s*([^\s:*"]+|"[^"]+"))?/giu;
+const filterRegex = /(\b(AND|OR)\s+)?(-?)(([\w]+|"[^"]+")\s*(:\s*([^\s:*"]+|"[^"]+"))|(\b[^:\s"]+\b|"[^"]+"))/giu;
+// Relevant groups
+// 2 - relation
+// 3 - negation?
+// 5 - filterName?
+// 7 - filterValue
+// 8 - quoted/unquoted Text
+
+
+interface Term {
+  relation: string;
+  value: string;
+}
+
+const trimQuotes = (str: string) => str.startsWith('"') ? str.substr(1, str.length - 2) : str;
+
+module.exports = (searchString: string): any => {
```

Comment:
Should `export default function(.....`

Reply:
Also please define the structure that's being returned as TypeScript interface. In general, we should avoid `any` types.